### ISSUE TITLE
bug when using not operator as prefix of a stream operation

### ISSUE THREAD

**Issue by MichelMunoz** [nushell/nushell#6581]
### Describe the bug

while exploring negation in nushell, I encountered a probable bug.

### How to reproduce

```
> '/home/joe/test.txt' | str ends-with '.txt'
true
> '/home/joe/test.txt' | not str ends-with '.txt'
true
```
the first is correct, the second not, at least for a normal user.

### Expected behavior

either a parse error ("not" not allowed there) or the correct boolean result, depending on the specs of not in nushell

### Configuration

nushell 0.68.1
linux

### Additional context

It was discovered while playing with negation, particularly _"do something if file/path do not exist"_, there does not seems to have an elegant solution :
`somepath | not path exists` <-- **not possible**
`somepath | path exists | not ` <-- **not possible**
the only way I found is `somepath | path exists | not $in` which is not in line with other nore "natural" possibilities like `somelist | is-empty`

**Comment by sholderbach** [comment]
Sadly a variety of commands are not composable like that yet. The `not` command works in the context of expressions (like math and boolean checks). The string/path commands on the other hand are operating only as the first command in a pipeline step. This also includes the boolean check commands. Currently the work around in expression syntax would be `somepath | not ($in | path exists)

**Comment by MichelMunoz** [comment]
Yes, I guessed so, but at least we should have a parsing error for `<stuff> | not <some other command>`

My intuition, is that it is related to a more general concept : so far (I'm playing with nushell on real cases to see pain points and limitations), it seems that nushel has two "worlds"
- functional (call by providing all arguments)
- stream (pipes as one implicit argument)

And, as far as I know either a command is for function call (with all mandatory args provided) or pipe call, there seems not to be a "**dual** definition".
That would be possible with something like below, where an explicit argument would receive $in if that argument is not provided:
```
def myfunction(main_arg : sometype = $sin, secondary_arg) [    <--- note the default value of main_arg !!
  do stuff
]
```
such a construct would allow a command to be called in function mode
`myfunction(arg1, arg2)`
and in pipe mode
`arg1 | myfunction arg2`

For this ticket, `not` could thus work in pipe or function call type ; but that would be true for many other functions.

My apologies if this is not very clear, I hope you see the idea, it's just foof for thought. This kind of dual declaration should probably be limited to fixed number of args (otherwise some semantic problems arise).
I might create a question ticket about that idea later when I had more time to think about it.

**Comment by sholderbach** [comment]
At the moment we don't have a notion of function call syntax ( the `($in | path contains)`  in my example is a [subexpression](https://www.nushell.sh/book/variables_and_subexpressions.html#subexpressions) that happens to look like an argument to the unary operator not.)
But I personally agree with you that it would be valuable if our commands could behave consistent with respect to the input stream or argument.
This would kind of follow [universal function call syntax](https://en.wikipedia.org/wiki/Uniform_Function_Call_Syntax) or how pipeline operators in other functional style languages work (e.g.  F#, Haskell [Flow](https://hackage.haskell.org/package/flow-2.0.0.1/docs/Flow.html), R's `magrittr` and now core, or the proposed C++ syntax)

**Comment by MichelMunoz** [comment]
Yeah, exactly... but... at least for nushell, you would want an "explicit" syntax (such as in my explanation), because some nushell commands should net be "dual" (or universal) ; only explicitly desgnated commands should, 
eg. 
`where` is probably not a good candidate for universal call
but `length` is a good candidate (`x | length` or `length <some list>`)
